core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java

```diff
@@ -138,13 +149,19 @@ public RegistryService getRegistryService() {
                 encryptedKey = encryption.encryptWithSymmetricKey(plainTextBytes);
             } else {
                 Cipher keyStoreCipher;
+                KeyStore keyStore;
+                Certificate[] certs;
                 KeyStoreManager keyMan = KeyStoreManager.getInstance(
                         MultitenantConstants.SUPER_TENANT_ID,
                         this.getServerConfigService(),
                         this.getRegistryService());
-                KeyStore keyStore = keyMan.getPrimaryKeyStore();
-
-                Certificate[] certs = keyStore.getCertificateChain(keyAlias);
+                if (keyMan.getInternalKeyStore() != null) {
```

Comment:
Is it possible for keyMan object to be null? If so, need to have a null check

Reply:
No, it is not possible to be a null value. KeyStoreManager.getInstance will return new KeyStore object.